src/reseeding.rs

```diff
@@ -122,7 +122,8 @@ pub struct ReseedWithNew;
 #[cfg(feature="std")]
 impl<R: Rng + NewSeeded> Reseeder<R> for ReseedWithNew {
     fn reseed(&mut self, rng: &mut R) {
-        match R::new() {
+        // TODO: should we use new_with_fallback instead?
```

Comment:
Reseeding a strong RNG like ISAAC from something much weaker like the system timer does not sound like a good idea.

`ReseedingRng` wraps an already seeded RNG, so we only have to care about reseeding, not initializing.

In some situations it makes sense to ignore an error, like with `TreadRng`. Instead of taking down a process, it seems preferable to continue using the existing RNG. But someone relying on reseeding to keep the wrapped RNG secure would probably like to be informed of an error. It should be passed on through `try_fill`, and otherwise panic.

So a best default is not clear to me. Maybe we should just do the simplification of `TreadRng` to not rely on `ReseedingRng`, and handle errors different for them.

In both cases I don't think `new_with_fallback` is a good idea.

Reply:
If a `ReseedingRng` was initially seeded from the OS *then* this failed when reseeding, then yes, it would be better to keep using the old state or mix in new entropy than replace. But the only case I have heard of `OsRng` failing is that "broken DLL" problem on Windows (i.e. never works) or (possibly) slow initialisation (doesn't work at first, does later).

We could in any case always mix in fresh entropy instead of replace. I don't know.

As for "not seeding a strong RNG like ISAAC from a weak source", why not? It's still better than seeding a weak RNG from a weak source. As that bug report about Firefox shows, sometimes a fallback option is needed, but there's no reason they need to use a weak RNG too.